lib/cloud/index.js

```diff
@@ -1,139 +1,101 @@
-    console.log(err);
-    throw new Error(err);
-  }
-});
-
-function writeStreamToFile(fileMeta, stream) {
-  var deferred = q.defer();
-  stream.on('end', function() {
-    // console.log('File save complete:', filename);
-    deferred.resolve(fileMeta);
-  });
-  stream.on('error', function(error) {
-    deferred.reject(error);
-  });
-  var filename = imageDir + '/' + fileMeta.uid;
-  stream.pipe(fs.createWriteStream(filename));
-  return deferred.promise;
-}
-
-var parseBase64Stream = function(req) {
-  var passthrough = false;
-  var accumulation = '';
-  var stream = req.pipe(through(function(chunk, enc, callback) {
-    if (!passthrough) {
-      accumulation += chunk;
-      var test = ';base64,';
-      var index = accumulation.indexOf(test);
-      if (index > - 1) {
-        passthrough = true;
-        chunk = accumulation.substr(index + test.length);
-      }
-    }
-    if (passthrough) {
-      this.push(chunk);
-    }
-    callback();
-  }))
-    .pipe(base64.decode());
-  return stream;
-};
-
-
-function initRouter(mediator) {
-  var router = express.Router();
+function initRouter(mediatorService) {
+  var router = express.Router()
+  
   router.route('/all').get(function(req, res) {
-    mediator.publish(config.cloudDataTopicPrefix + config.datasetId + ':list');
-    mediator.once('done:' + config.cloudDataTopicPrefix + config.datasetId + ':list', function(files) {
-      res.json(files);
+    mediatorService.listFilesMetadata(null).then(function(fileList) {
+      res.json(fileList);
     });
   });
-
+  
   router.route('/owner/:owner').get(function(req, res) {
-    mediator.publish(config.cloudDataTopicPrefix + config.datasetId + ':list');
-    mediator.once('done:' + config.cloudDataTopicPrefix + config.datasetId + ':list', function(files) {
-      // TODO send filter to storage engine directly instead of filtering large results in code
-      // Requires change in
-      var filtered = files.filter(function(file) {
-        return String(file.owner) === String(req.params.owner);
-      });
-      res.json(filtered);
+    var owner = req.params.owner;
+    mediatorService.listFilesMetadata(owner).then(function(fileList) {
+      res.json(fileList);
     });
   });
-
+  
   router.route('/owner/:owner/upload/base64/:filename').post(function(req, res, next) {
     var uid = uuid.create().toString();
     var fileMeta = {
       owner: req.params.owner,
       name: req.params.filename,
-      id: uid,
-      uid: uid
+      namespace: req.params.namespace,
+      uid: uid,
+      id: uid
     };
-    var stream = parseBase64Stream(req);
-    // TODO store files in gridfs
-    writeStreamToFile(fileMeta, stream).then(function(fileMeta) {
-      return mediator.request(config.cloudDataTopicPrefix + config.datasetId + ':create', fileMeta);
-    })
-    .then(function(fileMeta) {
+    var stream = fileService.parseBase64Stream(req);
+    var promise = fileService.writeStreamToFile(fileMeta, stream);
```

Comment:
Should `promise` be called `writeBase64ToStream` maybe ? `promise` is a bit generic.